OneCommunityGlobal/HighestGoodNetworkApp
# Nathan separate reports from weekly summaries

# Description
This PR separates the permissions for seeing the weekly summaries and reports under the reports tab.

## Related PRS (if any):
This frontend PR is related to the [#643](https://github.com/OneCommunityGlobal/HGNRest/pull/643) backend PR.

## Main changes explained:
- Add permisison check for 'getReports' in header to access the reports page
- Limit 'getWeeklySummaries' to only the weekly summaries page

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log in as users with each combination of the `getReports` and `getWeeklySummaries` permissions (see notes)
5. go to the dashboard
6. Check if the reports tab appears if you have either `getWeeklySummaries` or `getReports` permissions
7. Check if 'Reports' appears in the tab if you have `getReports`

## Screenshots or videos of changes:
Both permissions:
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/23086054/12e473ac-02c3-43ab-88da-27dd9f94c7f3)
Only reports:
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/23086054/1aa939bd-45db-46ca-964c-ee69cbc087ae)

Only summaries:
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/23086054/0a2a38bc-eae4-4ba9-a207-9e316011e8c6)

Neither permission (no reports tab):
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/23086054/7f7cfd86-48a6-4042-a31b-08ac0c33993f)

## Note:
You can check what permission your user has with redux dev tools under:
1. `state.auth.user.permissions.frontPermissions` for individual permissions
2. `state.role.roles.{the role of your account}` for role permissions

If you need to assign permissions to a user, log in with an Owner account and go to Other Links->Permission Management->Manage User Permissions and give/remove needed permissions from a volunteer account

On the permissions management page, `getReports` is labeled 'See Analytics Reports' and `getWeeklySummaries` is labeled 'See Weekly Summaries'.
